### PR TITLE
Adds a PULL_REQUEST_TEMPLATE and accompanying CHANGELOG file

### LINKED ISSUE #22: Community standards: Add PR templates
Likewise, having a PR template also facilitates providing all the relevant information about what is being added and encourage the submitted to review if it ticks all the relevant boxes. Again, check PyBaMM for inspiration:

[PyBaMM/PULL_REQUEST_TEMPLATE.md at develop · pybamm-team/PyBaMM (github.com)](https://github.com/pybamm-team/PyBaMM/blob/develop/.github/PULL_REQUEST_TEMPLATE.md)

More on Community Standards: [Community Standards (github.com)](https://github.com/ImperialCollegeLondon/virtual_rainforest/community)

### PR DESCRIPTION
**Pull request by TaranRallings**
What other aspects of our workflow need to be worked into the "Key Checklist"?

Does CHANGELOG need a specific structure at this point?

fixes #22

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# [Unreleased](https://github.com/ImperialCollegeLondon/virtual_rainforest/)
```
Comment (alexdewar) [inline CHANGELOG.md:1]: I'm not sure why you've added an empty changelog here...? Maybe you guys do want to have a discussion about having a changelog at some point but:

1. Adding it should really be a separate PR (or at least a separate git commit!)
2. I'd wait until you've got something to put in it
3. I'd open raise it as an issue first

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Please add a line in the relevant section of [CHANGELOG.md](https://github.com/ImperialCollegeLondon/virtual_rainforest/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.
+
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Optimization (back-end change that speeds up the code)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+
+
+# Key checklist:
+
+- [ ] No style issues: `$ flake8`
```
Comment (alexdewar) [inline PULL_REQUEST_TEMPLATE.md:18]: As we're using ``pre-commit`` hooks, I'd tell users to make sure they've run those (which includes a hook for ``flake8``) instead.
```suggestion
- [ ] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
```

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] All tests pass: `$ python run-tests.py --unit`
```
Comment (alexdewar) [inline PULL_REQUEST_TEMPLATE.md:19]: There is no file `run-tests.py`. To run the tests, users will need to run `pytest` instead.

File: PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`
+
+You can run all three at once, using `$ python run-tests.py --quick`.
```
Comment (alexdewar) [inline PULL_REQUEST_TEMPLATE.md:22]: See comment above. I'd just drop this sentence as it doesn't apply to your repo

**Review by alexdewar — CHANGES_REQUESTED** [review]
Hi @TaranRallings.

I've got some suggestions and comments for you.

One other small thing is: you either want to have "closes #issue" or "fixes #issue" in the description so that GitHub can automatically detect that it actually fixes the issue.

**Comment by alexdewar** [comment]
As before, it'd be great for someone else from the project to offer their review (@davidorme?) as long as you're not snowed under with GitHub emails :laughing:

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (dalonsoa) [inline PULL_REQUEST_TEMPLATE.md:19]: We don't run tests this way but with `pytest`:

```suggestion
- [ ] All tests pass: `$ poetry run pytest`
```

File: PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (dalonsoa) [inline PULL_REQUEST_TEMPLATE.md:18]: Probably we want to ask if all `pre-commit` hooks pass, not just `flake8`. If `pre-commit` is properly setup and the changes were made locally, this is automatically fulfilled but not if they changes were made directly in GitHub. So, something along these lines will be better:

```suggestion
- [ ] No style issues: `$ poetry run pre-commit run --all-files`
```

File: PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (dalonsoa) [inline PULL_REQUEST_TEMPLATE.md:22]: This is not correct for our repo. Remove.

**Review by dalonsoa — CHANGES_REQUESTED** [review]
There are some specific things that need adapting to this project.

File: CHANGELOG.md (hunk 1 above)
Comment (TaranRallings) [reply CHANGELOG.md:1]: It seemed to be connected to the recommended template, that was my only motivation. I'll remove it.

**Comment by TaranRallings** [comment]
I got turned around by clicking "commit changes" from the suggested changes, realizing I needed to make other changes on my laptop, and then getting stuck on merge conflicts. Rerequesting review.

**Review by dalonsoa — APPROVED** [review]

**Review by alexdewar — APPROVED** [review]
